azurerm/internal/services/apimanagement/api_management_api_resource.go

```diff
@@ -175,6 +175,53 @@ func resourceArmApiManagementApi() *schema.Resource {
 				Default:  false,
 			},
 
+			"oauth2_authorization": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"authorization_server_name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.ApiManagementChildName,
+						},
+						"scope": {
+							Type:     schema.TypeString,
+							Optional: true,
+							// There is currently no validation, as any length and characters can be used in the field
+						},
+					},
+				},
+			},
+
+			"openid_authentication": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"openid_provider_name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.ApiManagementChildName,
+						},
+						"bearer_token_sending_methods": {
+							Type:         schema.TypeSet,
```

Comment:
Should this be a set?  Thinking over it, there are only 2 possible values, would it not be more appropriate to have these as individual bools?